### DISCUSSION TITLE
Repo Structures

### DISCUSSION THREAD

**Discussion by HSorensen** [GCC-Tiny/.github#5] [General]
We should remove the unneeded repos:

- https://github.com/GCC-Tiny/gcc-tiny-doc
- https://github.com/GCC-Tiny/gccpl1
- https://github.com/GCC-Tiny/gcc-tiny

The Tiny source is in the branch **src-gcc-tiny** of the https://github.com/GCC-Tiny/gcc repo.

Thoughts?

**Comment by SamuraiCrow** [comment]
I wish it didn't need a massive GCC monorepo but maybe it does.

**Reply by HSorensen** [reply]
If we want to track latest gcc, I think this is the only way.  We can look at git modules, but I think the separate repo is much easier. I have no strong preference, beside ease of maintenance...
There is of course the possibility ... ( long shot) ... we would upstream tiny to the gcc project :)

**Reply by HSorensen** [reply]
Also, as we explore more advanced features of GCC like right now I am researching enabling the tests etc. To enable this there are changes needed in the gcc-src/Makefile.def so one way or another we need a clone of the repo.

**Reply by SamuraiCrow** [reply]
Ok.

**Reply by SamuraiCrow** [reply]
I've deleted the old Markdown docs experiment and the unmodified GCC-Tiny sources to avoid confusion and pinned their replacements.  I've left PL1 for now since it can be an example of how to use ANTLR with GCC.

**Comment by HSorensen** [comment]
Just realized we should not push directly to the src-gcc-tiny branch. We should only use PRs. I protected the branch accordingly.

**Reply by SamuraiCrow** [reply]
I think we are the only 2 people with commit privilages so far but without a review procedure in place, it just adds another step.

**Reply by HSorensen** [reply]
yes, but the point is to have the gcc branch track upstream, and src-gcc-tiny track changes for the tiny language, so that branch is more like a clone of the gcc project. So I would prefer we use PR to review changes instead of just keep committing directly to the branch.

**Reply by SamuraiCrow** [reply]
Ok.  It seems you're doing the heavy lifting in this project anyway.  No harm?  No foul.
